### PR TITLE
Added unzipping strategy

### PR DESCRIPTION
**Pull request by dolonm-mw**
Changed the unzip strategy for genscript resources.

### CODE REVIEW

**Review by mcafaro — CHANGES_REQUESTED** [review]

**Review by nbhoski — CHANGES_REQUESTED** [review]
I see 1161 files changed. Can you please revert that last commit and resubmit

File: src/test/java/com/mathworks/ci/RunMatlabTestsBuilderTest.java [hunk 1]
```diff
@@ -1,12 +1,14 @@
 package com.mathworks.ci;
```
Comment (mcafaro) [inline src/test/java/com/mathworks/ci/RunMatlabTestsBuilderTest.java:1]: This file only appears to have whitespace changes. Can you remove it from the change list?

File: src/main/resources/com/mathworks/ci/RunMatlabTestsBuilder/config.jelly [hunk 2]
```diff
@@ -17,14 +17,15 @@
 <f:block>
```
Comment (mcafaro) [inline src/main/resources/com/mathworks/ci/RunMatlabTestsBuilder/config.jelly:17]: This file only appears to have whitespace changes. Can you remove it from the change list?

File: src/main/java/com/mathworks/ci/RunMatlabTestsBuilder.java [hunk 3]
```diff
@@ -273,14 +273,17 @@ private int execMatlabCommand(FilePath workspace, Launcher launcher,
```
Comment (mcafaro) [inline src/main/java/com/mathworks/ci/RunMatlabTestsBuilder.java:273]: This file only appears to have whitespace changes. Can you remove it from the change list?

File: src/main/java/com/mathworks/ci/MatlabBuilderConstants.java [hunk 4]
```diff
@@ -1,4 +1,5 @@
-package com.mathworks.ci;
+       package com.mathworks.ci;
```
Comment (mcafaro) [inline src/main/java/com/mathworks/ci/MatlabBuilderConstants.java:1]: Something strange going on with indentation here.

File: src/main/java/com/mathworks/ci/MatlabBuild.java [hunk 5]
```diff
@@ -67,6 +70,8 @@ default void copyFileInWorkspace(String sourceFile, String targetFile, FilePath
         targetFilePath.copyFrom(in);
         // set executable permission
         targetFilePath.chmod(0755);
+     
+
```
Comment (mcafaro) [inline src/main/java/com/mathworks/ci/MatlabBuild.java:74]: Try to avoid unnecessary whitespace changes. They make it more difficult to see where the actual code changes are. If possible, try to get the review down to just the lines where there are actual code changes (unless the existing whitespace needs to be fixed).

File: src/main/java/com/mathworks/ci/MatlabBuild.java [hunk 6]
```diff
@@ -81,29 +86,33 @@ default FilePath getFilePathForUniqueFolder(Launcher launcher, String uniqueName
     default String getUniqueNameForRunnerFile() {
         //Using 8 bit long random alphanumeric string
         return RandomStringUtils.randomAlphanumeric(8);
-    }
+    } 
     
     // This method prepares the temp folder by coping all helper files in it.
-    default void prepareTmpFldr(FilePath tmpFldr, String runnerScript) throws IOException, InterruptedException {
+       default void prepareTmpFldr(FilePath tmpFldr, String runnerScript) throws IOException, InterruptedException {
```
Comment (mcafaro) [inline src/main/java/com/mathworks/ci/MatlabBuild.java:92]: Something strange going on with indentation here.

File: src/main/java/com/mathworks/ci/MatlabBuild.java [hunk 7]
```diff
@@ -81,29 +86,33 @@ default FilePath getFilePathForUniqueFolder(Launcher launcher, String uniqueName
...
-                MatlabBuilderConstants.MATLAB_SCRIPT_GENERATOR, tmpFldr);
-        FilePath zipFileLocation =
-                new FilePath(tmpFldr, MatlabBuilderConstants.MATLAB_SCRIPT_GENERATOR);
-
-        // Unzip the file in temp folder.
-        zipFileLocation.unzip(tmpFldr);
-    }
+ }
     
+    default String getZipScript(String script, String url) {
```
Comment (mcafaro) [inline src/main/java/com/mathworks/ci/MatlabBuild.java:105]: `getZipScript` seems like a strange name for this function. It seems like this function is just replacing a placeholder in the a string. Perhaps something like `replaceZipPlaceholder`?

File: src/main/java/com/mathworks/ci/MatlabBuild.java [hunk 8]
```diff
@@ -81,30 +82,33 @@ default FilePath getFilePathForUniqueFolder(Launcher launcher, String uniqueName
...
+                 new FilePath(tmpFldr, MatlabBuilderConstants.MATLAB_SCRIPT_GENERATOR);
+         runnerScript=replaceZipPlaceholder(runnerScript, zipFileLocation.getRemote());
+         
+      // Write MATLAB scratch file in temp folder.
+         FilePath scriptFile =
+                 new FilePath(tmpFldr, getValidMatlabFileName(tmpFldr.getBaseName()) + ".m");
+         scriptFile.write(runnerScript, "UTF-8");
     }
     
+    default String replaceZipPlaceholder(String script, String url) {
```
Comment (nbhoski) [inline src/main/java/com/mathworks/ci/MatlabBuild.java:102]: it would be good practice to add some comments for new methods describing what it does

**Review by nbhoski — CHANGES_REQUESTED** [review]

File: src/main/resources/com/mathworks/ci/RunMatlabTestsBuilder/config.jelly [hunk 9]
```diff
@@ -24,7 +24,7 @@
     </f:optionalBlock>
     
     <f:optionalBlock name="tapArtifact" field="tapArtifact" title="TAP test results" checked="${instance.tapArtifact.selected}">   	
-     <f:entry field="tapReportFilePath" title="File path: ">	
+     <f:entry field="tapReportFilePath" title="File path: ">
```
[the remark this answers is not in the document]
Comment (dolonm-mw) [reply src/main/resources/com/mathworks/ci/RunMatlabTestsBuilder/config.jelly:27]: Whitespace change only

File: src/main/java/com/mathworks/ci/MatlabBuilderConstants.java [hunk 10]
```diff
@@ -37,14 +38,22 @@ public class MatlabBuilderConstants {
...
+			+ "addpath(destination);\n"
+			+"zipURL='${ZIP_FILE}';\n"
+			+"unzip(zipURL,destination);\n"
+			+"disp('Unziping genscript resource in');\n"
+			+"disp(destination);\n"
+			+ "testScript = genscript(${PARAMS});\n"
+	        + "disp('Running MATLAB script with content:');\n"
+	        + "disp(strtrim(testScript.Contents));\n"
+	        + "runnerFile = testScript.writeToFile(fullfile(destination,'runnerScript.m'));\n"
+	        +"warning('off');"
```
[the remark this answers is not in the document]
Comment (dolonm-mw) [reply src/main/java/com/mathworks/ci/MatlabBuilderConstants.java]: err11
To remove this line from console output. But it seems like a way to display the temp directory to the user. Hence removed.

**Review by nbhoski — APPROVED** [review]
Ship it !

File: src/main/java/com/mathworks/ci/MatlabBuild.java [hunk 11]
```diff
@@ -85,26 +86,30 @@ default String getUniqueNameForRunnerFile() {
...
+                 MatlabBuilderConstants.MATLAB_SCRIPT_GENERATOR, tmpFldr);
+         FilePath zipFileLocation =
+                 new FilePath(tmpFldr, MatlabBuilderConstants.MATLAB_SCRIPT_GENERATOR);
+         runnerScript=replaceZipPlaceholder(runnerScript, zipFileLocation.getRemote());
+         
+      // Write MATLAB scratch file in temp folder.
+         FilePath scriptFile =
+                 new FilePath(tmpFldr, getValidMatlabFileName(tmpFldr.getBaseName()) + ".m");
+         scriptFile.write(runnerScript, "UTF-8");
     }
```
Comment (mcafaro) [inline src/main/java/com/mathworks/ci/MatlabBuild.java:100]: Still something off with indentation in these files. Looks like a mixture of tabs and spaces are being used.

File: src/main/java/com/mathworks/ci/MatlabBuild.java [hunk 12]
```diff
@@ -85,23 +85,27 @@ default String getUniqueNameForRunnerFile() {
...
-        zipFileLocation.unzip(tmpFldr);
+        // Write MATLAB scratch file in temp folder.
+        FilePath scriptFile =
+                new FilePath(tmpFldr, getValidMatlabFileName(tmpFldr.getBaseName()) + ".m");
+        scriptFile.write(runnerScript, "UTF-8");
+    }
+
+    //This method replaces the placeholder with genscript's zip file location URL in temp folder
+    default String replaceZipPlaceholder(String script, String url) {
+        script = script.replace("${ZIP_FILE}", url);
```
Comment (mcafaro) [inline src/main/java/com/mathworks/ci/MatlabBuild.java:103]: Any single quotes in `url` need to be escaped because it is being put inside single quotes in the test runner script text.
